### ISSUE TITLE
Enhancement: Request remote monitor for all rigs through web page

### ISSUE THREAD

**Issue by Byrd1083** [RainbowMiner/RainbowMiner#58]
Is there a feature similar to multipoolminer monitor? Would like to load a chrome window from my phone remotely to see profitability on all rigs running.

**Comment by RainbowMiner** [comment]
I will see, what I can do.

RBM

**Comment by RainbowMiner** [comment]
There is a reporting service build into RainbowMiner. A report url and api key can be set for that.
Do you have you own webserver where you want to monitor your rigs?
- I can provide some php source code, so that you could use it like that
 Or do you prefer using some service running on another webserver?
- you still can use something like multipoolminer.io
RBM

**Comment by Byrd1083** [comment]
Yes, maybe include some setup instructions for using miltipoolminer monitor with this. I have been happy with that monitor. I already have a key I've been using.

**Comment by Byrd1083** [comment]
It is just nice when I'm on the road to be able to pull up a web page from my phone to view the current stats of all miners in one window. It is a fast way to see if one went down as well. Beats trying to remote login which works okay with chrome remote desktop but can be buggy and unreliable at times and won't get you anywhere if a miner goes down. If I didn't have the monitor, I would not no if a miner crashed or just had bad internet service.

**Comment by RainbowMiner** [comment]
Easy: edit config.txt and enter your data
  "MinerStatusURL": "https://multipoolminer.io/monitor/miner.php",
  "MinerStatusKey": "<put your status key here>",

RBM

**Comment by giannibonfanti** [comment]
Hi @RainbowMiner,
I have developed a cloud monitoring platform (for mining rigs) that maybe can interest you.
It already supports MultiPoolMiner reporting service, and it's used by some clients I have.
It extends and improves the functionalities provided by multipoolminer.io, like realtime email notifications, average profit calculations, and it's architecture is ready to accomodate more types of opensource mining software helpers. We are working on a mobile app and push notifications.

Are you interested in integrate this into RainbowMiner?
Basically it's just another API call into ReportStatus.ps1

If this it's something you would like to add to RainbowMiner, I'm interested to collaborate with you on this.
I think it's very useful to have a dashboard of all your rigs on your phone and receive notifications when a rig goes offline/online.

Regards,

G

**Comment by RainbowMiner** [comment]
> Are you interested in integrate this into RainbowMiner?

Yes, that sounds interesting.

> If this it's something you would like to add to RainbowMiner, I'm interested to collaborate with you on this.

Ok, tell me more.

RBM

**Comment by giannibonfanti** [comment]
As an example, I can show you the dashboard page:

![dashboard](https://user-images.githubusercontent.com/19237095/45785552-0f5cd780-bc6d-11e8-89b0-f576553d2850.png)

This is currently done using default data sent from MPM ReportStatus.ps1 script (slightly modified).
Other detailed data are shown on each miner info box.

Moreover, RainbowMiner handles more useful data, as temps, power, OC profiles, ecc.. if useful we can add them accordingly and show them on the page (and trigger notifications).

There is a setting page where to define worker names and per-workers notifications, as well as few profile info (eg: timezone).

I can tell you more about the infrastructure and how to send data to our api: can I reach you out by email or how? Mine is giannibonfa [at] hotmail [dot] com

**Comment by TheeeX** [comment]
Hi, I am not able to see Remote workers, 
Clients show that it's connected to the server, but the Remote Workers tab is not visible on the server dashboard.
Also If I open http://localhost:4000/clients on the server machine, it shows all the clients.
Also, I can see the worker and server at https://rbminer.net

What could be the issue?

**Comment by TheeeX** [comment]
> There is a reporting service build into RainbowMiner. A report url (=MinerStatusUrl) and api key (=MinerStatusKey) can be set for that in the config.txt.
> 
> Do you have you own webserver where you want to monitor your rigs?
> 
>     * I can provide some php source code, so that you could use it like that
>       Or do you prefer using some service running on another webserver?
> 
>     * you still can use something like multipoolminer.io
> 
> 
> RBM

Can you open-source the code for RainbowMiner monitoring central (rbminer.net).

**Comment by TheeeX** [comment]
@giannibonfanti, @Byrd1083  Do you have implementation of RainbowMiner monitoring central that you can share ?

**Comment by RainbowMiner** [comment]
@TheeeX Hi, sorry, it seems I have missed your request. What do you want to implement? If you can line it out a bit, I can extract the necessary code snippets.

**Comment by TheeeX** [comment]
I basically want to see live profits of my workers and also make them visible on my RainbowMiner Server.

to make this happen I think i would require API for:

1. Client Report Api + Server response
2. Api that server uses to show remote workers on dashboard

**Comment by TheeeX** [comment]
@RainbowMiner , @Byrd1083 can you please email the code. connect.theeex@gmail.com

**Comment by RainbowMiner** [comment]
@TheeeX Ok, I'll extract the important code parts.

**Comment by currentemailname** [comment]
@RainbowMiner @giannibonfanti @Byrd1083 @sxb1n9 @TheeeX 

Hello, can anyone please share monitoring central or similar dashboard source code / snippets.

**Comment by RainbowMiner** [comment]
Sorry, I still had no time to extract the code snippets, yet. I will take care of it as soon as possible.

**Comment by currentemailname** [comment]
> Sorry, I still had no time to extract the code snippets, yet. I will take care of it as soon as possible.

Hello @RainbowMiner, Any luck finding time to extract the code snippets 🤞 🙇
